src/substrate/filters/enricher.ts

```diff
@@ -30,6 +30,29 @@ export async function Enrich(
     excludeAddresses?: string[],
   }> => {
     switch (kind) {
+      /**
+       * ImOnline Events
+       */
+      case EventKind.HeartbeatReceived: {
+        const [ authorityId ] = event.data as unknown as [ AuthorityId ] & Codec
+        return {
+          data: {
+            kind,
+            authorityId: authorityId.toString()
+          }
+        }
+      }
+      case EventKind.SomeOffline: {
+        const [ validators ] = event.data as unknown as [ Vec<IdentificationTuple> ];
+        const sessionIndex = await api.query.session.currentIndex();
+        return {
+          data: {
+            kind,
+            sessionIndex: +sessionIndex -1,
```

Comment:
Why -1?

Reply:
According to my understanding, SomeOffline Event gets fired at the end of a session when any validator was found to be offline, and enricher gets executed after that (where a new session have begun);
I intend to indicate that `validators` were offline in the `sessionIndex`; To give the actual sessionIndex where `validators` were offline, I have introduced -1

Reply:
Cool -- can you add something brief and explanatory in a comment? That would be useful for the future. Thanks!